internal/cmd/gocdk/main.go

```diff
@@ -46,53 +46,35 @@ func main() {
 	err = run(ctx, pctx, os.Args[1:], &debug)
 	done()
 	if err != nil {
-		if debug {
-			fmt.Fprintf(os.Stderr, "%+v\n", err)
-		} else {
-			// TODO(light): format error message parts one per line?
-			fmt.Fprintf(os.Stderr, "%v\n", err)
-		}
-		if xerrors.As(err, new(usageError)) {
-			os.Exit(64)
-		}
 		os.Exit(1)
 	}
 }
 
 func run(ctx context.Context, pctx *processContext, args []string, debug *bool) error {
-	globalFlags := newFlagSet(pctx, "gocdk")
-	globalFlags.BoolVar(debug, "debug", false, "show verbose error messages")
-	if err := globalFlags.Parse(args); xerrors.Is(err, flag.ErrHelp) {
-		return nil
-	} else if err != nil {
-		return usagef("gocdk: %w", err)
-	}
-	if globalFlags.NArg() == 0 {
-		return usagef("gocdk COMMAND ...")
+	var rootCmd = &cobra.Command{
+		Use:   "gocdk",
+		Short: "TODO gocdk",
+		Long:  "TODO more about gocdk",
+		// We want to print usage for any command/flag parsing errors, but
+		// suppress usage for random errors returned by a command's RunE.
+		// This function gets called right before RunE, so suppress from now on.
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			cmd.SilenceUsage = true
+		},
 	}
+	rootCmd.PersistentFlags().BoolVar(debug, "debug", false, "show verbose error messages")
```

Comment:
IIUC, this flag doesn't do anything any more, so we could remove it.

Cobra has a `SilenceErrors` setting that we could use, and then print out the errors ourselves (including using the `debug` flag for verbose errors), but that has the down side of printing out errors after usage for cases where usage is printed, which IMHO looks weird.

Reply:
That's unfortunate. The `%+v` output has been really useful to me for understanding failures, and I think will be even more so when we need to collect errors from folks in the field. I'd be okay removing it as long as we have an issue to track somehow regaining this functionality.

Reply:
Filed https://github.com/google/go-cloud/issues/2173 and remove `--debug`.